source/Conversation.cpp

```diff
@@ -188,14 +187,14 @@ void Conversation::Load(const DataNode &node, const string &missionName)
 		{
 			// This is just an ordinary text node.
 			// If the previous node is a choice, or if the previous node ended
-			// in a goto, create a new node. Otherwise, just merge this new
-			// paragraph into the previous node.
-			if(nodes.empty() || !nodes.back().canMergeOnto)
+			// in a goto, or if the new node has a condition, then create a new
+			// node. Otherwise, just merge this new paragraph into the previous
+			// node.
+			if(nodes.empty() || !nodes.back().canMergeOnto || HasCondition(child))
```

Comment:
I dont really know why HasCondition has to be a function, as it is only called once.
I am not particular against it, but I want to know why.

Reply:
It makes the code easier to read, especially since actually answering the question "does the given `DataNode` have a condition?" requires a for loop.

Reply:
What if we just remove the need for a loop by using `child.HasChildren()`? I would expect any node that has children of it's own not to be mergeable onto the previous node.

Reply:
I can't off the top of my head think of a reason why that wouldn't work, but something is tickling the back of my mind, saying there's some vanilla case that will be screwed up by that.

Reply:
Right, it messes up the reload on the last goto-label in a savegame (that text-line would not be merged onto the previous text node while this code would merge it)...

![last_goto_label](https://user-images.githubusercontent.com/35403542/202308883-cb0a78fd-1923-4ea1-b9c3-7e4eaa646216.png)